Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/AddUserController.java

```diff
@@ -18,35 +25,54 @@
 @Controller
 public class AddUserController {
 
-	@Autowired
-	IUserManager usersManager;
-
-	/**
-	 * This method creates a new user for given user information
-	 * 
-	 * @param req
-	 *            Holds HTTP request information
-	 * @return String to redirect user to user list page
-	 */
-	@RequestMapping(value = "auth/user/createuser")
-	public String createUser(HttpServletRequest req) {
-
-		User user = new User(req.getParameter("username"),
-				req.getParameter("password"));
-		user.setName(req.getParameter("fullname"));
-		user.setEmail(req.getParameter("email"));
-		usersManager.addUser(user);
-		return "redirect:/auth/user/list";
-	}
-
-	/**
-	 * This method provides string to redirect user to add user page
-	 * 
-	 * @return String to redirect user to add user page
-	 */
-	@RequestMapping(value = "auth/user/add")
-	public String addUser() {
-		return "auth/user/add";
-	}
+    @Autowired
+    IUserManager usersManager;
+    @Autowired
+    UserValidator uValidator;
+
+    @InitBinder
+    private void initBinder(WebDataBinder binder) {
+        binder.setValidator(uValidator);
+    }
+
+    /**
+     * This method creates a new user for given user information
+     * 
+     * @param req
+     *            Holds HTTP request information
+     * @param model
+     *            Holds generic information about the model
+     * @param user
+     *            Holds the User bean values retrieved from the add page
+     * @param result
+     *            Binding Result Object for binding errors if any
+     * @return String to redirect user to user list page
+     * 
+     */
+    @RequestMapping(value = "auth/user/createuser")
+    public String createUser(HttpServletRequest req, ModelMap model, @ModelAttribute("user") @Validated User user,
+            BindingResult result) {
+        if (result.hasErrors()) {
+            return "auth/user/add";

```

Comment:
I'm not 100% sure, but I believe you have to add the user to the model to show it on the add page.